### PR TITLE
added server-side userlist pagination and many prettiefies

### PR DESCRIPTION
**Pull request by Demengjkee**

### CODE REVIEW

**Comment by kmarmaliykov** [comment]
backend changes are ok

File: frontend/app/scripts/controllers/trainingCtrl.js [hunk 1]
```diff
@@ -5,13 +5,19 @@ angular.module('frontendApp')
     'UserFeedbackService',
     function ($route, $scope, $location, $localStorage, ngDialog, training, UserFeedbackService) {
 
+
+      $scope.isMentor = function() {
```
Comment (pashapovalov) [inline frontend/app/scripts/controllers/trainingCtrl.js]: May be the 'isTrainer' name better?

File: frontend/app/scripts/controllers/userlistCtrl.js [hunk 2]
```diff
@@ -2,9 +2,28 @@
 
 
 angular.module('frontendApp')
-  .controller('UserlistCtrl', ['$scope', 'ngDialog', 'data', 'userService', '$location',
-    function ($scope, ngDialog, data, userService, $location) {
-      $scope.users = data;
+  .controller('UserlistCtrl', ['$scope', 'ngDialog',  'userService', '$location', 'userServiceDelegate', '$http',
+    function ($scope, ngDialog,    userService, $location, userServiceDelegate, $http) {
+      //$scope.users = data;
```
Comment (pashapovalov) [inline frontend/app/scripts/controllers/userlistCtrl.js]: please remove this comment

File: frontend/app/views/userprofile.html [hunk 3]
```diff
@@ -54,9 +43,23 @@
     </div>
 
     <div class="col-md-3 watchUserFeedback">
-      <button class="btn btn-primary" ng-click="wathchFeedback()">{{translation.WATCH_USER_FEEDBACKS}}</button>
+      <button ng-if="isAdmin()" class="btn btn-primary" ng-click="wathchFeedback()">
+        {{translation.WATCH_USER_FEEDBACKS}}</button>
+      <div class="btn-user-profile-edit" ng-if="isAdmin() && isUserExternal">
+        <a class="btn btn-primary btn-sm" ng-click="startEdit()"
```
Comment (pashapovalov) [inline frontend/app/views/userprofile.html]: update formatting please

``` html
<a class="btn btn-primary btn-sm" ng-click="startEdit()" ng-show="!editMode">
     <i class="fa fa-pencil"></i> {{translation.EDIT}}
</a>
```

File: frontend/app/views/userprofile.html [hunk 4]
```diff
@@ -54,9 +43,23 @@
...
+           ng-show="!editMode">
+          <i class="fa fa-pencil"></i> {{translation.EDIT}}</a>
+        <a class="btn btn-primary btn-sm" ng-click="saveChanges()"
+           ng-show="editMode">
```
Comment (pashapovalov) [inline frontend/app/views/userprofile.html]: and this, too

File: frontend/app/views/tabs/trainingEntriesTab.html [hunk 5]
```diff
@@ -1,3 +1,4 @@
+<div style="margin: 0 250px 0 250px">
```
Comment (pashapovalov) [inline frontend/app/views/tabs/trainingEntriesTab.html]: No style in html!

**Pull request merged**